test.functional/build.gradle.kts

```diff
@@ -1,7 +1,62 @@
+import com.moowork.gradle.node.yarn.YarnTask
+import java.util.function.Consumer
+
 plugins {
     id("java")
+    id("com.cognifide.aem.instance")
+    id("com.moowork.node")
 }
 
 description = "Example - Functional Tests"
 
+tasks{
+    register<YarnTask>("testFunctional") {
+        dependsOn("testSetup", "testInstallNode")
+        group = "check"
+        setYarnCommand("jest")
+        setWorkingDir(file("${projectDir.absolutePath}/test.functional"))
+        outputs.dir(file("${projectDir.absolutePath}/test.functional"))
+        doFirst {
+            val props = mutableListOf()
+            if (project.hasProperty("aem.env")) {
+                props.addAll(listOf("--config", "./env/${project.properties["aem.env"]}.config.js"))
+            } else {
+                props.addAll(listOf("--config", "./env/local-publish.config.js"))
+            }
+            args = props
+        }
+    }
+
+    register("testSetup") {
+        description = "Creates env json from gradle properties."
```

Comment:
describe more general context like `Generates environment related configuration for functional tests`

how about name `setupFunctional` task name? `test` prefix is usually reserved for running tests

Reply:
Done